src/main.rs

```diff
@@ -127,6 +133,18 @@ struct Cli {
     /// Deprecated. Print all log information to stdout
     #[arg(long)]
     log_all: bool,
+    /// API key to be supplied to the endpoint, if supported.
+    #[cfg(feature = "search")]
+    #[arg(long, default_value = "")]
+    api_key: String,
+    /// The URL for the LlamaEdge query server. Supplying this implies usage.
+    #[cfg(feature = "search")]
+    #[arg(long, required = true)]
+    query_server_url: String,
+    /// The search API backend to use for internet search.
+    #[cfg(feature = "search")]
+    #[arg(long, default_value = "tavily", requires = "query-server-url")]
+    search_backend: String,
```

Comment:
1. How many possible values for the `search_backend` argument?
2. If the possible values are known, then it would be better to define an Enum type (for example, `SearchBackend`) and make each possible value a variant; and then, annotate each variant with `#[value(name = "...")]`. After that, the `search_backend` argument can be declared as `SearchBackend` type; and `--help` will show the possible values. Here is an example as a reference:

```rust
use clap::ValueEnum;

#[derive(Clone, Debug, Copy, PartialEq, Eq, ValueEnum)]
pub enum SearchBackend {
        #[value(name = "tavily")]
        Tavily,
        ...
}
```

Reply:
The search backends available to the server depend on the llamaedge query server being used. By default, it comes with 2 (Tavily and Bing), but new backends can be added by whoever is hosting the llamaedge-query-server. 

As such, I decided to simply use a string since that doesn't put a limit on what backends the server has available.

Reply:
Got it, thanks!